pay-api/migrations/versions/2024_03_06_04b8a7bed74e_ejv_link_generic.py

```diff
@@ -0,0 +1,32 @@
+"""empty message
+
+Revision ID: 04b8a7bed74e
+Revises: bacb2b859d78
+Create Date: 2024-03-06 11:40:00.153387
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '04b8a7bed74e'
+down_revision = 'bacb2b859d78'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('ejv_invoice_links', sa.Column('link_type', sa.String(length=20), nullable=False))
+    op.drop_index(op.f('ix_ejv_invoice_links_invoice_id'), table_name='ejv_invoice_links')
+    op.alter_column('ejv_invoice_links', 'invoice_id', new_column_name='link_id',
```

Comment:
Would it still be useful to index the link_id for querying purposes since we used to have one for invoice_id?

Reply:
the link id is now become either refund_partial_id or invoice_id, instead, I'm adding link_type column to indicate it's point to refund_partial or invoice

Reply:
The index will be on link_id value itself, it won't need to be tied to specifically invoice_id or the refund_partial_id. It will probably be good as I believe the data warehouse queries use the links so having an index would speed up the joins.

Maybe even an index for link_type + link_id.

Reply:
done, added `op.create_index('ix_ejv_links_link_type_link_id', 'ejv_links', ['link_type', 'link_id'], unique=False)`